plugins/ingest/src/main/java/org/elasticsearch/ingest/IngestDocument.java

```diff
@@ -231,7 +242,7 @@ private static Object resolve(String pathElement, String fullPath, Object contex
      * @throws IllegalArgumentException if the path is null, empty, invalid or if the field doesn't exist.
      */
     public void appendFieldValue(String path, Object value) {
-        setFieldValue(path, value, true);
+        setFieldValue(TemplateService.wrap(path), ValueSource.wrap(value), true);
```

Comment:
> does this mean that a path provided as string doesn't support a template?

No. If processors impl like to support templates they can can use the method variant that does support templating. I the case of this method, we just need to add `public void appendFieldValue(Template path, ValueSource value) {...}` because no processor is using it yet. Or better just change to current method to accept Template and ValueSource, since I think the `push` processor will also support templating.